### PR TITLE
Add reservoir level tests related to delayed initialization and heuristic parameter

### LINKED ISSUE #134: Add end-to-end test for new reservoir levels
#125

### PR DESCRIPTION
**Pull request by flomnes**
Fixes #134

### CODE REVIEW

File: src/tests/examples/example_test.py [hunk 1]
```diff
@@ -37,6 +37,13 @@ def find_integrity_path(output_dir):
     assert len(op) == 1
     return op[0]
 
+def find_values_hourly_path(output_dir):
```
Comment (jmkerloch) [inline src/tests/examples/example_test.py:40]: This function could be used for a new test. We can add to parameters : 
- area 
- year

File: src/tests/examples/example_test.py [hunk 2]
```diff
@@ -87,6 +94,15 @@ def check_integrity_second_opt(path):
 
     np.testing.assert_allclose(reference_values[4:8], output_values[4:8], rtol=1e-4, atol=0)
 
+def fetch_hourly_hydro_reservoir_values(path):
```
Comment (jmkerloch) [inline src/tests/examples/example_test.py:97]: We could have a more common function `fetch_hourly_values` that will take the column to be read as a parameter

File: src/tests/examples/example_test.py [hunk 3]
```diff
@@ -530,6 +546,38 @@ def test_075_kcg_on_four_areas_02(use_ortools, ortools_solver, solver_path):
...
+    reservoir_levels = fetch_hourly_hydro_reservoir_values(study_path)
+    first_march_midnight = 1417
+    assert(abs(reservoir_levels[first_march_midnight] - 30) < .05)
+
+@pytest.mark.short
+def test_hydro_hydro_preference_1(use_ortools, ortools_solver, solver_path):
+    study_path = ALL_STUDIES_PATH / "short-tests" / "hydro preference 1"
+    run_study(solver_path, study_path, use_ortools, ortools_solver)
+    reservoir_levels = fetch_hourly_hydro_reservoir_values(study_path)
+    assert(abs(reservoir_levels[-1] - 30.46) < .05)
```
Comment (jmkerloch) [inline src/tests/examples/example_test.py:572]: What's the behavior of `reservoir_levels[-1]` ? All values are tested ?

**Review by jmkerloch — CHANGES_REQUESTED** [review]
Refactor some test functions to be used for other tests.

File: src/tests/examples/example_test.py (hunk 3 above)
Comment (flomnes) [reply src/tests/examples/example_test.py:572]: `array[-1]` refers to the last coefficient of a Python list/array

File: src/tests/examples/example_test.py (hunk 1 above)
Comment (flomnes) [reply src/tests/examples/example_test.py]: Done

File: src/tests/examples/example_test.py (hunk 2 above)
Comment (flomnes) [reply src/tests/examples/example_test.py]: Done

**Review by jmkerloch — APPROVED** [review]

**Pull request merged**